src/mpm/termlibrary3.C

```diff
@@ -71,8 +71,10 @@ void TMBTSigTerm::computeTMgeneralizedStrain (FloatArray& answer, FloatMatrix& B
     // evaluate temperature at given point
     this->temperatureField->interpolation->evalN(Nt, lcoords, FEIElementGeometryWrapper(&cell));
     double t = Nt.dotProduct(rt);
-    answer.append(gradT); // construct generalized strain vector
-    answer.append(t); // add temperature
+    answer.resize(Bu.size()+gradT.size()+1);
```

Comment:
Just to discuss: Seems that append can be supported as (pseudocode):

def append (array):
resize.(this->size+array->size)
this->copySubVector(array, ....)

For me keeping append contributes to code readability, as the method does exactly what is supposed to do, while the purpose of suggested code to replace it is hard to guess on first sight. For appending a single vector, there is no overhead, appending multiple vectors can be done using std::list as in other places.
What do you think?

Reply:
I replaced that one (and a few others) with a new `FloatArray::fromConcatenated({vec1,vec2,...})`, that is quite legible. `append` needs to resize, hence re-allocate amd copy old data over (Eigen has no concept of `reserve`, unlike `std::vector`).